frontend/static/js/app.js

```diff
@@ -1,37 +1,89 @@
-Vue.component('correction-view', {
-    props: ['token'],
-    data: function () {
-        return {
-            message: 'Loading correction...',
-            correction: null,
-        }
+Vue.component("correction-view", {
+  props: ["token"],
+  data: function () {
+    return {
+      message: "Loading correction...",
+      correction: null,
+      originalHadith: { data: null, errors: [] },
+    };
+  },
+  created: function () {
+    this.getNextCorrection();
+  },
+  updated: function () {
+    if (
+      this.correction &&
+      this.originalHadith.data === null &&
+      !this.originalHadith.loading &&
+      this.originalHadith.errors.length === 0
+    ) {
+      let correction = this.correction;
+      this.fetchOriginalHadith(correction.id);
+    }
+  },
+  methods: {
+    getNextCorrection: function () {
+      fetch("/corrections", {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `token ${this.token}`,
+        },
+      })
+        .then((resp) =>
+          resp.ok
+            ? resp
+            : (function () {
+                throw new Error(`Http status ${resp.status}`);
+              })()
+        )
+        .then((resp) => resp.json())
+        .then((data) => {
+          if (!data || data.length == 0) alert("no more corrections");
+          else {
+            this.message = "loaded";
+            this.correction = data[0];
+          }
+        })
+        .catch((err) => alert("something went wrong " + err.message));
     },
-    created: function () {
-        this.getNextCorrection();
+    fetchOriginalHadith: function (hadithUrn) {
+      this.message = "Loading original hadith...";
+      this.originalHadith.loading = true;
+      fetch("/hadtihs/" + hadithUrn, {

```

Comment:
the code seems to be repeated, can we factor out common parts of fetch usage to a separate function and use it in all places?